### PR TITLE
Refactor: Apply extract ids for search results

### PR DESCRIPTION
**Pull request by glensc**
Continue https://github.com/glensc/python-pytrakt/pull/9

### CODE REVIEW

**Comment by glensc** [comment]
I don't really understand what the `extract_ids` do, as code before and after continues to work. but I would like the code to be consistent.

next step is to use MixIn to simplify things:
- https://github.com/moogar0880/PyTrakt/pull/186

File: trakt/sync.py [hunk 1]
```diff
@@ -288,19 +288,26 @@ def search_by_id(query, id_type='imdb', media_type=None, slugify_query=False):
         if 'episode' in d:
             from trakt.tv import TVEpisode
             show = d.pop('show')
-            extract_ids(d['episode'])
+            episode = d.pop('episode')
+            extract_ids(episode)
             results.append(TVEpisode(show.get('title', None),
                                      show_id=show['ids'].get('trakt'),
```
Comment (twolaw) [inline trakt/sync.py:294]: For consistancy, you should extract show ids just like in users.py

**Review by twolaw — CHANGES_REQUESTED** [review]

File: trakt/sync.py (hunk 1 above)
Comment (glensc) [reply trakt/sync.py:294]: what do you mean? using `item['type']`?

File: trakt/users.py [hunk 2]
```diff
@@ -132,27 +132,28 @@ def get_items(self):
...
+                extract_ids(item_data)
+                episode = TVEpisode(show_data['title'], **item_data, show_id=show_data['trakt'])
                 self._items.append(episode)
             elif item_type == 'person':
-                self._items.append(Person(item_data['name'],
-                                          item_data['slug']))
+                extract_ids(item_data)
+                self._items.append(Person(**item_data))
 
         yield self._items
```
Comment (glensc) [inline trakt/users.py:158]: I noticed that `UserList.get_items()` has a bug, it always appends items to `self._items`. IMHO it should reset `self._items = []` on each call to `get_items()`.

File: trakt/sync.py (hunk 1 above)
Comment (twolaw) [reply trakt/sync.py:294]: extract_ids(show)

show_id=show.get('trakt')

Comment (glensc) [reply trakt/sync.py]: This then: ea690b64ecfed63354a225840d097a9a591635ca

**Comment by glensc** [comment]
Probably useless after https://github.com/glensc/python-pytrakt/pull/14

**Pull request closed without merging**